obspy/io/reftek/core.py

```diff
@@ -315,12 +311,25 @@ def to_stream(self, network="", location="", component_codes=None,
                                                              encoding)
                         elif encoding in ('16', '32'):
                             dtype = {'16': np.int16, '32': np.int32}[encoding]
+                            bytes_ = {'16': 2, '32': 4}[encoding]
                             # just fix endianness and use correct dtype
                             sample_data = np.require(
                                 packets_['payload'],
                                 requirements=['C_CONTIGUOUS'])
+                            # account for number of samples
+                            sample_bytes = sample_data.shape[1]
+                            sample_data = sample_data.flatten()
+                            nbr_samples_max = sample_bytes // bytes_
+                            inds_ = [ind for ind, p in
+                                     enumerate(packets_["number_of_samples"])
+                                     if p != nbr_samples_max]
+                            for ind in inds_[::-1]:
+                                start = ind * sample_bytes + bytes_ * packets_["number_of_samples"][ind]  # noqa: E501
+                                end = (ind + 1) * sample_bytes
+                                sample_data = np.delete(sample_data,
+                                                        np.arange(start, end))
```

Comment:
This could be more idiomatic. It took me a while to understand this list a build and then reverse iteration.
You using `'number_of_samples'` again and then looking up
```suggestion
                            for ind, num_samps in reversed([
                                    (ind, num_samps) for ind, num_samps in
                                    enumerate(packets_["number_of_samples"])
                                    if num_samps != nbr_samples_max]):
                                start = ind * sample_bytes + bytes_ * num_samps
                                end = (ind + 1) * sample_bytes
                                sample_data = np.delete(sample_data,
                                                        np.arange(start, end))
```

Reply:
I think this logic would also simplify a bit if done after the appropriate conversion to 16 or 32-bit ints, wouldn't have to multiply by sample size